swap_meet/vendor.py

```diff
@@ -1,2 +1,229 @@
+    #             return thing
+    #     return False
+
+    def get_by_category(self, category):
+        itemList = []
+        
+        for item in self.inventory:
+            if item.category == category:
+                itemList.append(item)
+        #returns a list of `Item`s in the inventory with that category
+        return itemList
+
+    # using list comprehension, don't have to use append
+    # itemList = [item for item in self.inventory if item.category == category]
+    """
+    
+    To further reduce the amount of repeated code in your project, 
+    consider how `swap_best_by_category` and `swap_first_item` might be able to make use of `swap_items`. 
+    Is there a way that these methods could incorporate a call to `swap_items` into the body of these methods?
+
+    ###To do this, could create default paremeters or is it keyword arguments for my_item and their item in swap items
+    then create an if, elif situation in swap items, then replace the expression in the if statement of swap_first_item with
+    the swap_items function ???? not sure what the default parameters/keyword args would be tho...
+    update: Oh my gosh I don't know why I thought about this like this, can just use in the other funcs like Hailey and Philomena did....
+    
+    """    
+    
+    def swap_items(self,friend, my_item, their_item):
+        #If this `Vendor`'s inventory doesn't contain `my_item` or the friend's inventory doesn't contain `their_item`, the method returns `False`
+        if my_item not in self.inventory or their_item not in friend.inventory:
+            return False
+
+        #remove the `my_item` from this `Vendor`'s inventory, and add it to the friend's inventory
+        my_removed = self.remove(my_item)
+        friend.inventory.append(my_removed)
+
+        #remove the `their_item` from the other `Vendor`'s inventory, and add it to this `Vendor`'s inventory
+        their_removed = friend.remove(their_item)
+        self.inventory.append(their_removed)
+        
+        return True
+
+    def swap_first_item(self,friend):
+        #If either itself or the friend have an empty `inventory`, the method returns `False`
+
+        #can use not like Anya did because I think empty lists are falsey
+        #if not self.inventory or not friend.inventory:
+        if len(self.inventory) == 0 or len(friend.inventory) == 0:
+            return False
+
+        #remove the first item from its `inventory`, and adds the friend's first item
+        #remove the first item from the friend's `inventory`, and adds the instances first item
+
+        #need to add first to both and then remove otherwise it won't work...
+        #can just use self.swap_items and pass in the first items of each list
+        self.add(friend.inventory[0])
+        friend.add(self.inventory[0])
+        
+        self.remove(self.inventory[0])
+        friend.remove(friend.inventory[0])
+        
+        return True
```

Comment:
We can also utilize `swap_items()` as a helper method.

Reply:
Currently, swapping any items has a time complexity of O(n) due to the `remove` method in `swap_items`. We can make a subtle improvement to O(1) by using multiple assignment instead:

```python
self.inventory[0], other.inventory[0] = other.inventory[0], self.inventory[0]
```        

Here's more info on Python multiple assignment: https://www.w3schools.com/python/gloss_python_assign_value_to_multiple_variables.asp